### ISSUE TITLE
C API sample not compilable

### ISSUE THREAD

**Issue by xykong58** [microsoft/onnxruntime#1843]
**Describe the bug**

**Urgency**
If there are particular important use cases blocked by this or strict project-related timelines, please share more information and dates. If there are no hard deadlines, please specify none.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- ONNX Runtime installed from (source or binary):
- ONNX Runtime version:
- Python version:
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**To Reproduce**
Describe steps/code to reproduce the behavior:

**Expected behavior**

**Screenshots**

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

**Comment by xykong58** [comment]
Two issues,

- C API documentation is too rough, it is hard to follow.  A basic user's guide for using the API
  will help.

- The example for C API fails to compile.   Please provides compiling command line
   and other information which makes it at least compiling OK.

$ g++ --version
g++ (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


$ g++ -std=c++11 -I./include c_sample.cpp

c_sample.cpp: In function ‘int main(int, char**)’:
c_sample.cpp:37:56: error: ‘ORT_ENABLE_BASIC’ was not declared in this scope
   OrtSetSessionGraphOptimizationLevel(session_options, ORT_ENABLE_BASIC);
                                                        ^
c_sample.cpp:62:60: error: ‘OrtGetAllocatorWithDefaultOptions’ was not declared in this scope
   CHECK_STATUS(OrtGetAllocatorWithDefaultOptions(&allocator));
                                                            ^
c_sample.cpp:15:31: note: in definition of macro ‘CHECK_STATUS’
     OrtStatus* onnx_status = (expr);                     \
                               ^
c_sample.cpp:130:3: error: ‘OrtMemoryInfo’ was not declared in this scope
   OrtMemoryInfo* memory_info;
   ^
c_sample.cpp:130:18: error: ‘memory_info’ was not declared in this scope
   OrtMemoryInfo* memory_info;
                  ^
c_sample.cpp:131:89: error: ‘OrtCreateCpuMemoryInfo’ was not declared in this scope
   CHECK_STATUS(OrtCreateCpuMemoryInfo(OrtArenaAllocator, OrtMemTypeDefault, &memory_info));
                                                                                         ^
c_sample.cpp:15:31: note: in definition of macro ‘CHECK_STATUS’
     OrtStatus* onnx_status = (expr);                     \
                               ^
c_sample.cpp:137:35: error: ‘OrtReleaseMemoryInfo’ was not declared in this scope
   OrtReleaseMemoryInfo(memory_info);
                                   ^
c_sample.cpp:34:50: warning: ignoring return value of ‘OrtStatus* OrtSetSessionThreadPoolSize(OrtSessionOptions*, int)’, declared with attribute warn_unused_result [-Wunused-result]
   OrtSetSessionThreadPoolSize(session_options, 1);
                                                  ^
c_sample.cpp:94:78: warning: ignoring return value of ‘OrtStatus* OrtGetDimensions(const OrtTensorTypeAndShapeInfo*, int64_t*, size_t)’, declared with attribute warn_unused_result [-Wunused-result]
     OrtGetDimensions(tensor_info, (int64_t*)input_node_dims.data(), num_dims);
                                                                              ^

**Comment by xykong58** [comment]
I am using the latest release version of C API (version 0.5).

**Comment by RyanUnderhill** [comment]
@xykong58 I made a fix to the sample 5 days ago, so that should be good now. The documentation is the only remaining issue.
